src/client/Microsoft.Identity.Client/Utils/ManagedIdentityCertificateManager.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+
+namespace Microsoft.Identity.Client.Internal.Utilities
+{
+    internal static class ManagedIdentityCertificateManager
+    {
+        private const string CertificateName = "devicecert.mtlsauth.local";
+        private static readonly object s_lock = new object();
+        private static X509Certificate2 s_cachedCertificate;
+        private static bool s_isInMemoryCertificate;
+
+        /// <summary>
+        /// Event triggered when a new certificate is created.
+        /// </summary>
+        internal static event Action<X509Certificate2> CertificateUpdated;
+
+        /// <summary>
+        /// Main entry point for retrieving a certificate. Splits logic between Linux (always in-memory)
+        /// and Windows/macOS (attempt platform store first, else create in-memory).
+        /// </summary>
+        /// <param name="forceUpdate">If true, forces renewal of an in-memory certificate.</param>
+        /// <returns>An X509Certificate2 valid for 90 days.</returns>
+        public static X509Certificate2 GetOrCreateCertificate(bool forceUpdate = false)
+        {
+            lock (s_lock)
```

Comment:
Do we really need to block on reading the cert? Can you use a https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-threading-readerwriterlockslim instead?